### PR TITLE
Update devdeps

### PR DESCRIPTION
**Pull request by creviera**
# Description

Fixes 4 security alerts due to CVE-2022-42969: https://github.com/freedomofpress/securedrop-client/security

This only impacts our dev environment. One of the alerts was on a file that was no longer used, so I did a cleanup in the first commit.

The second commit updates `pytest` and `pytest-xdist` to address security alerts relating to CVE-2022-42969. See https://github.com/pytest-dev/py/issues/287#issuecomment-1290407715 for additional context.

# Test Plan

- [ ] CI passes
- [ ] No non-dev requirements files were changed

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [x] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration is [self-contained] and applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [x] No database schema changes are needed

[self-contained]: https://github.com/freedomofpress/securedrop-client#generating-and-running-database-migrations

### CODE REVIEW

**Comment by creviera** [comment]
whoops, I forgot to use the appropriate VMs for each requirements file... anyone have bookworm already?

File: requirements/dev-bookworm-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
+# This file is autogenerated by pip-compile with python 3.7
```
Comment (gonzalo-bulnes) [inline requirements/dev-bookworm-requirements.txt:2]: We've seen differences in at least indirect dependencies between Python 3.9 (default on Bullseye) and Python 3.10 (default on Bookworm), so it is best to generate each of these files with the corresponding Python version. (Python 3.10 for this one.)

(I don't know if that may work towards fixing the CI failures.)

File: requirements/dev-bullseye-requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.9
+# This file is autogenerated by pip-compile with python 3.7
```
Comment (gonzalo-bulnes) [inline requirements/dev-bullseye-requirements.txt:2]: Similarly, I'd re-generate this file with Python 3.9 before investigating further the CI failures 🙂

File: requirements/dev-sdw-requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.9
```
Comment (gonzalo-bulnes) [inline requirements/dev-sdw-requirements.txt:2]: Python 3.9 is our default now that we target Bullseye in production. As above, I'd prefer if this file was re-generated using Puthon 3.9.

File: requirements/dev-mac-requirements.in [hunk 4]
```diff
@@ -1,4 +1,4 @@
--r dev-requirements.in
+-r dev-sdw-requirements.in
```
Comment (gonzalo-bulnes) [inline requirements/dev-mac-requirements.in:1]: Shouldn't this be `dev-bullseye-requirements.in` (the dev requirements that match the production Python version)?

**Review by gonzalo-bulnes — CHANGES_REQUESTED** [review]
I think some of the Python bersioning might contribute to the CI failures. Even if not, I think it's worth generating the files with specific Python versions to limit unnecessary diffs durimg updates 🙂

**Comment by gonzalo-bulnes** [comment]
You outpaced my review @creviera ! A venv with Python 3.10 is enough, no real need for a Bookworm VM, but I can do that part if that helps too 🙂

**Comment by creviera** [comment]
I just tried manually updating the bookworm requirements comment to python 3.10 instead of python 3.9... we'll see if that does the trick

**Comment by creviera** [comment]
> You outpaced my review @creviera ! A venv with Python 3.10 is enough, no real need for a Bookworm VM, but I can do that part if that helps too slightly_smiling_face

@gonzalo-bulnes - yes, please! Thanks for the offer. If it's quick for you to do, it might make sense to append to my readme update so that we recommend using `dev-bullseye-requirements.txt` now + how to update dev deps by just using the correct virtual env (venv-bookworm with python3.10) etc.

File: requirements/dev-mac-requirements.in (hunk 4 above)
Comment (creviera) [reply requirements/dev-mac-requirements.in:1]: oops, you're right

**Comment by gonzalo-bulnes** [comment]
I was wondering about what the README said, indeed, but was away from a large screen. I'll re-generate the Bookworm requirements file (it does have a few differences with the Bullseye one, even if those are just comments) and take a look at the README. Thank you for opening this PR! :tada:

**Review by gonzalo-bulnes — APPROVED** [review]
The changes look goo to me now.

- I added the `py` requirement. The `pytest` change log does mention that might be necessary depending on specific use cases. (The `pytest` package itself doesn't depend on `py` anymore. Details in the commit messages.)
- I reworded the section of the README that touches on dependency updates.
- And I made sure the requirement files were generated by the corresponding version of Python, as the README now instructs :stuck_out_tongue: 

@creviera Do you want to take a look at the changes to cross-review them? :eyes:

File: README.md [hunk 5]
```diff
@@ -248,10 +248,28 @@ securedrop-client
 
 `dev-*-requirements.txt` and `requirements.txt` point to python software foundation hashes, and `build-requirements.txt` points to our builds of the wheels from our own pip mirror (https://github.com/freedomofpress/securedrop-debian-packaging/tree/main/localwheels). Whenever a dependency in `build-requirements.txt` changes, our team needs to manually review the code in the dependency diff with a focus on spotting vulnerabilities.
 
-If you're adding or updating a dependency, you need to:
+### Production
 
-1. Modify either `requirements.in` or `dev-*-requirements.in` (depending on whether it is prod or dev only)
-2. Run `make update-pip-requirements`. This will generate `dev-*-requirements.txt` and `requirements.txt`
+If you're adding or updating a production dependency, you need to:
```
Comment (creviera) [inline README.md:253]: New instructions lgtm!

**Review by creviera — COMMENTED** [review]
lgtm! your commit messages were helpful too. woohoo! :tada:

**Comment by gonzalo-bulnes** [comment]
(rebased, only changing one line in the README to fix the conflict)

**Review by gonzalo-bulnes — APPROVED** [review]
Based on @creviera and my own cross-reviews :slightly_smiling_face:

**Pull request merged**